sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -942,6 +977,47 @@ func withDynamicResolver(ctx context.Context, ref string) containerd.RemoteOpt {
 		}
 	default:
 		// For all other registries
-		return noOp
+		return defaultResolver
+	}
+}
+
+// registryHosts returns the registry hosts to be used by the resolver
+// Heavily borrowed from CRI plugin's implementation with the auth related configuration omitted.
+// See https://github.com/containerd/cri/blob/f6026296a3991010429db91e7e677f9c9d4861ab/pkg/server/image_pull.go#L314-L315
+// FIXME Replace this once there's a public containerd client interface that supports registry mirrors
+func registryHosts(registryConfig *RegistryConfig, authorizer docker.Authorizer) docker.RegistryHosts {
+	return func(host string) ([]docker.RegistryHost, error) {
+		var registries []docker.RegistryHost
+		var endpoints []string
+		_, ok := registryConfig.Mirrors[host]
+		if ok {
+			endpoints = registryConfig.Mirrors[host].Endpoints
+		} else {
+			endpoints = registryConfig.Mirrors["*"].Endpoints
+		}
```

Comment:
We only support a single wildcard match rather than globbing?

Reply:
Right, this would match up with containerd-cri's behavior. `"*"` here just means for all other hosts that weren't specified in the config, use `"*"`'s backup endpoints.